src/wp-includes/class-wp-theme-json.php

```diff
@@ -764,6 +764,18 @@ private function get_block_classes( $style_nodes ) {
 				}
 			}
 
+			/*
+			 * Reset default browser margin on the root body element.
+			 * This is set on the root selector **before** generating the ruleset
+			 * from the `theme.json`. This is to ensure that if the `theme.json` declares
+			 * `margin` in its `spacing` declaration for the `body` element then these
+			 * user-generated values take precedence in the CSS cascade.
+			 * @link https://github.com/WordPress/gutenberg/issues/36147.
+			 */
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
+				$block_rules .= "body { margin: 0; }\n";
```

Comment:
```suggestion
				$block_rules .= "body { margin: 0; }";
```

Removing the new line character to be consistent with other rules in the file and for phpunit tests to pass.